example/multipart/consumer/tests/Service/HttpClientServiceTest.php

```diff
@@ -0,0 +1,95 @@
+
+class HttpClientServiceTest extends TestCase
+{
+    public const URL_FORMAT = '^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()!@:%_\+.~#?&\/\/=]*)';
+
+    public function testUpdateUserProfile()
+    {
+        $matcher = new Matcher();
+        $fullName = 'Colten Ziemann';
+        $profileImageUrl = 'http://example.test/profile-image.jpg';
+        $personalNote = 'testing';
+
+        $request = new ConsumerRequest();
+        $request
+            ->setMethod('POST')
+            ->setPath('/user-profile')
+            ->setHeaders([
+                'Accept' => 'application/json',
+                'Authorization' => [
+                    \json_encode($matcher->like('Bearer eyJhbGciOiJIUzI1NiIXVCJ9'))
+                ],
+            ])
+            ->setBody(new Multipart([
+                new Part($fullNameTempFile = $this->createTempFile($fullName), 'full_name', 'text/plain'),
+                new Part(__DIR__ . '/../_resource/image.jpg', 'profile_image', in_array(php_uname('m'), ['AMD64', 'arm64']) ? 'application/octet-stream' : 'image/jpeg'),
+                new Part($personalNoteTempFile = $this->createTempFile($personalNote), 'personal_note', 'text/plain'),
+            ]));
+
+        $response = new ProviderResponse();
+        $response
+            ->setStatus(200)
+            ->addHeader('Content-Type', 'application/json')
+            ->setBody([
+                'full_name' => $matcher->like($fullName),
+                'profile_image' => $matcher->regex($profileImageUrl, self::URL_FORMAT),
+                'personal_note' => $matcher->like($personalNote),
+            ]);
+
+        $config = new MockServerConfig();
+        $config
+            ->setConsumer('multipartConsumer')
+            ->setProvider('multipartProvider')
+            ->setPactDir(__DIR__.'/../../../pacts');
+        if ($logLevel = \getenv('PACT_LOGLEVEL')) {
+            $config->setLogLevel($logLevel);
+        }
+        $builder = new InteractionBuilder($config);
+        $builder
+            ->given('User exists')
+            ->uponReceiving('A put request to /user-profile')
+            ->with($request)
+            ->willRespondWith($response);
+
+        $service = new HttpClientService($config->getBaseUri());
+        $userProfileResponse = $service->updateUserProfile();
+        $verifyResult = $builder->verify();
+
+        unlink($fullNameTempFile);
+        unlink($personalNoteTempFile);
+
+        $this->assertTrue($verifyResult);
+        $this->assertEquals([
+            'full_name' => $fullName,
+            'profile_image' => $profileImageUrl,
+            'personal_note' => $personalNote,
+        ], \json_decode($userProfileResponse, true, 512, JSON_THROW_ON_ERROR));
+    }
+
+    private function createTempFile(string $contents): string
+    {
+        $path = tempnam(sys_get_temp_dir(), 'pact');
+        //$newPath = "$path.txt";
+        //rename($path, $newPath);
```

Comment:
still need these comments or can they be removed?

Reply:
It's my bad to keep that comment there. It's better to mark this PR Draft. I will wait for 2 more fixes on pact-reference on multipart issues (one of them is https://github.com/pact-foundation/pact-reference/pull/319), then I will create new PR to delete this code completely (including this comment)

Reply:
Removed in https://github.com/pact-foundation/pact-php/pull/342